crates/networking/rpc/types/block_identifier.rs

```diff
@@ -64,15 +64,21 @@ impl BlockIdentifier {
             Ok(hex_str) => hex_str,
             Err(error) => return Err(RpcErr::BadParams(error.to_string())),
         };
-        // Check that the BlockNumber is 0x prefixed
-        let Some(hex_str) = hex_str.strip_prefix("0x") else {
-            return Err(RpcErr::BadHexFormat(arg_index));
+
+        if let Ok(value) = u64::from_str_radix(&hex_str, 10) {
+            return Ok(BlockIdentifier::Number(value));
+        };
+
+        let block_number_result = if let Some(hex_str_stripped) = hex_str.strip_prefix("0x") {
+            u64::from_str_radix(hex_str_stripped, 16)
+        } else {
+            u64::from_str_radix(&hex_str, 16)
         };
```

Comment:
Nit:
```suggestion
        let hex_str = hex_str.strip_prefix("0x").unwrap_or(hex_str);
        let block_number_result = u64::from_str_radix(hex_str, 16);
```

Reply:
Addressed in [358d7cc](https://github.com/lambdaclass/ethrex/pull/2335/commits/358d7ccf13d5ef1f3d3d32a8e71cacfe9902d36a) (the suggested commit did not quite work, it was missing an `&`)